src/ec2/model.ts

```diff
@@ -107,13 +115,13 @@ export class Ec2ConnectionManager {
     public async attemptEc2Connection(selection: Ec2Selection): Promise<void> {
         try {
             const response = await this.ssmClient.startSession(selection.instanceId)
+            const isConnectable = await this.isInstanceConnectable(selection.instanceId)
+            if (!isConnectable) {
+                throw new Error('Instance is not connectable.')
+            }
```

Comment:
Would this be user-facing? Should we use `ToolkitError` to show the user a more helpful message?

Reply:
This isn't user facing, this error is immediately caught and then passed into the `handleStartSessionError`. I imagine there this is a better way to handle this. Will try to refactor to make this flow clearer.